x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/IndicesAccessControl.java

```diff
@@ -21,11 +23,53 @@
  */
 public class IndicesAccessControl {
 
-    public static final IndicesAccessControl ALLOW_ALL = new IndicesAccessControl(true, Collections.emptyMap());
-    public static final IndicesAccessControl ALLOW_NO_INDICES = new IndicesAccessControl(true,
-            Collections.singletonMap(IndicesAndAliasesResolverField.NO_INDEX_PLACEHOLDER,
-                    new IndicesAccessControl.IndexAccessControl(true, new FieldPermissions(), DocumentPermissions.allowAll())));
-    public static final IndicesAccessControl DENIED = new IndicesAccessControl(false, Collections.emptyMap());
+    public static final IndicesAccessControl ALLOW_ALL = new IndicesAccessControl(true, null) {
+        @Override
+        public IndexAccessControl getIndexPermissions(String index) {
+            assert false == Regex.isSimpleMatchPattern(index) :
+                    "index access control can and should only be retrieved for a given concrete name, but requested [" + index + "]";
+            return IndexAccessControl.PROMISCUOUS_ACCESS_CONTROL;
+        }
+    };
+    public static final IndicesAccessControl ALLOW_NO_INDICES = new IndicesAccessControl(true, null) {
+        @Override
+        public IndexAccessControl getIndexPermissions(String index) {
+            for (String placeHolderIndexName : IndicesAndAliasesResolverField.NO_INDICES_OR_ALIASES_ARRAY) {
+                if (placeHolderIndexName.equals(index)) {
+                    return IndexAccessControl.PROMISCUOUS_ACCESS_CONTROL;
```

Comment:
This branch is accessed in request interceptors only, not in the `SecurityIndexReaderWrapper` or the security field filter.